bot/cogs/candy_collection.py

```diff
@@ -0,0 +1,185 @@
+            and the user reacting is not a bot (ie. ourselves)
+            Check if the reaction is a skull/candy first. """
+        message = reaction.message
+        if message.channel.id != self.HACKTOBER_CHANNEL_ID:
+            return
+        if user.bot:
+            return
+        if str(reaction.emoji) not in ('\N{SKULL}', '\N{CANDY}'):
+            if message.id in await self.ten_recent_msg():
+                await self.reacted_msg_chance(message)
+            return
+        for react in self.msg_reacted:
+            # check to see if the message id of a message we added a
+            # reaction to is in json file, and if nobody has won/claimed it yet
+            if react['msg_id'] == message.id and react['won'] is False:
+                react['user_reacted'] = user.id
+                react['won'] = True
+                try:
+                    # if they have record/candies in json already it will do this
+                    user_records = self.get_candyinfo[user.id]
+                    if str(reaction.emoji) == '\N{CANDY}':
+                        user_records['record'] += 1
+                    if str(reaction.emoji) == '\N{SKULL}':
+                        if user_records['record'] <= 3:
+                            user_records['record'] = 0
+                            lost = 'all of your'
+                        else:
+                            lost = random.randint(1, 3)
+                            user_records['record'] -= lost
+                        await self.send_spook_msg(message.author, message.channel, lost)
+
+                except KeyError:
+                    # otherwise it will raise KeyError so we need to add them
+                    if str(reaction.emoji) == '\N{CANDY}':
+                        print('ok')
+                        d = {"userid": user.id, "record": 1}
+                        self.candy_json['records'].append(d)
+                await self.remove_reactions(reaction)
+
+    async def reacted_msg_chance(self, message):
+        """(Randomly) add a skull or candy to a message if there is a reaction there already
+             (higher probability)"""
+        if random.randint(1, self.ADD_SKULL_EXISTING_REACTION_CHANCE) == 1:
+            d = {"reaction": '\N{SKULL}', "msg_id": message.id, "won": False}
+            self.msg_reacted.append(d)
+            return await message.add_reaction('\N{SKULL}')
+
+        if random.randint(1, self.ADD_CANDY_EXISTING_REACTION_CHANCE) == 1:
+            d = {"reaction": '\N{CANDY}', "msg_id": message.id, "won": False}
+            self.msg_reacted.append(d)
+            return await message.add_reaction('\N{CANDY}')
+
+    async def ten_recent_msg(self):
+        """Get the last 10 messages sent in the channel"""
+        ten_recent = []
+        recent_msg = max((x for x in self.bot._connection._messages
+                          if x.channel.id == self.HACKTOBER_CHANNEL_ID), key=lambda x: x.id)
```

Comment:
this max is quite icky.

simply `max(message.id for message in self.bot._connection._messages if message.channel.id == self.HACKTOBER_CHANNEL_ID)` would be better.